### PR TITLE
Implements API to NewsFeed & UserProfile Activities

### PR DESCRIPTION
**Pull request by lybaokhanh**
**Notes**: the branch need to **refactor code**, that's why I create a new pull request

Work in progress:

- [x] 1. NewsFeed
-- 1.1. News: CR
-- 1.2. Likes: CR
-- 1.3. Comment: CR

- [x] 2. UserProfile: Activities
- [x] 3. Upload Photos
-- 3.1.Load photos from local library
-- 3.2. Upload to AWS
- [x] 4. Take picture from Camera
-- 4.1. Capture screen
-- 4.2. Upload to AWS

### CODE REVIEW

File: App/Component/News/Comments/index.js [hunk 1]
```diff
@@ -1,20 +1,48 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { View, TextInput, TouchableOpacity } from 'react-native';
+import { graphql, compose } from 'react-apollo';
+import gql from 'graphql-tag';
```
Comment (pvtri96) [inline App/Component/News/Comments/index.js]: gql can be import within react-apollo

`import { gql } from 'react-apollo`

File: App/Component/News/index.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { compose, graphql } from 'react-apollo';
+import gql from 'graphql-tag';
```
Comment (pvtri96) [inline App/Component/News/index.js]: Same here

`import { gql } from 'react-apollo`

File: App/Scene/NewsFeed/index.js [hunk 3]
```diff
@@ -1,31 +1,101 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { FlatList } from 'react-native';
-import { graphql } from 'react-apollo';
+import { graphql, compose } from 'react-apollo';
```
Comment (pvtri96) [inline App/Scene/NewsFeed/index.js]: And here

`import { gql } from 'react-apollo`

File: App/Scene/Profile/Body/Content/Activities/index.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { News, LoadingIndicator } from '~/Component';
 
 import { graphql } from 'react-apollo';
 import gql from 'graphql-tag';
```
Comment (pvtri96) [inline App/Scene/Profile/Body/Content/Activities/index.js]: `import { gql } from 'react-apollo`

File: App/Scene/NewsFeed/index.js [hunk 5]
```diff
@@ -72,16 +137,55 @@ NewsFeedScene.footer = {
   activeColor: Colors.primary,
 };
 
-const NewsFeedSceneWithData = graphql(gql(NEWS_FEED_QUERY), {
-  props: ({
-    data: { loading, getAllNews, refetch, networkStatus, error },
-  }) => ({
+const MeQuery = graphql(gql(ME_QUERY), {
```
Comment (pvtri96) [inline App/Scene/NewsFeed/index.js]: `ME_QUERY` should be `QUERY_ME`

File: App/Scene/NewsFeed/index.js [hunk 6]
```diff
@@ -72,16 +137,55 @@ NewsFeedScene.footer = {
...
+  props: ({ data: { loading, me } }) => ({
     loading,
+    me,
+  }),
+});
+
+const AllNewsQuery = graphql(gql(ALL_NEWS_QUERY), {
```
Comment (pvtri96) [inline App/Scene/NewsFeed/index.js]: `QUERY_ALL_NEWS`

File: App/Scene/Profile/Body/Content/Activities/index.js [hunk 7]
```diff
@@ -48,20 +55,19 @@ Activities.propTypes = {
   allNews: PropTypes.array,
   refetch: PropTypes.func,
   networkStatus: PropTypes.number,
-  error: PropTypes.object,
+  user: PropTypes.object,
 };
 
 const ActivitiesWithQuery = graphql(gql(ACTIVITIES_QUERY), {
```
Comment (pvtri96) [inline App/Scene/Profile/Body/Content/Activities/index.js]: `QUERY_ACTIVITIES`

**Review by pvtri96 — CHANGES_REQUESTED** [review]

File: App/Component/News/Posts/index.js [hunk 8]
```diff
@@ -13,45 +14,84 @@ import {
   AutoExpandingTextInput,
 } from '~/Component';
 
-const defaultAvatar = Images.avatar['male08'];
+const ImagePickerConfig = {
+  allowsEditing: true,
+  aspect: [4, 3],
```
Comment (pvtri96) [inline App/Component/News/Posts/index.js]: Why a news photo must have aspect of 4:3 ?

Comment (lybaokhanh) [reply App/Component/News/Posts/index.js]: Because we should use the same resolution for every picture upload to server

**Review by pvtri96 — APPROVED** [review]

**Comment by pvtri96** [comment]
Hi @littlecrow, please resolve conflicts!

**Pull request merged**